node-feature-discovery-0.17.yaml

```diff
@@ -91,18 +83,44 @@ update:
 
 test:
   pipeline:
-    - runs: |
-        kubectl-nfd -h
-        nfd-gc -version
-        nfd-master -version
-        nfd-topology-updater -version
-        nfd-worker -version
-        kubectl-nfd --help
-        nfd-gc --version
-        nfd-gc --help
-        nfd-master --version
-        nfd-master --help
-        nfd-topology-updater --version
-        nfd-topology-updater --help
-        nfd-worker --version
-        nfd-worker --help
+    - uses: test/kwok/cluster
+    - name: "Test nfd-master startup"
+      uses: test/daemon-check-output
+      with:
+        setup: |
+          # Create minimal config file
+          mkdir -p /etc/kubernetes/node-feature-discovery
+          cat > /etc/kubernetes/node-feature-discovery/nfd-master.conf <<EOF
+          {
+            "noPublish": true,
+            "instance": "test",
+            "enableLeaderElection": false,
+            "nfdApiParallelism": 1,
+            "resyncPeriod": "1s",
+            "inClusterConfig": true,
+            "verifyServiceAccount": false
+          }
+          EOF
+
+          # Create minimal service account directory
+          mkdir -p /var/run/secrets/kubernetes.io/serviceaccount/
+          echo "fake-token" > /var/run/secrets/kubernetes.io/serviceaccount/token
+          echo "fake-ca" > /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+          echo "default" > /var/run/secrets/kubernetes.io/serviceaccount/namespace
+        start: env KUBERNETES_SERVICE_HOST=localhost KUBERNETES_SERVICE_PORT=8080 nfd-master --config=/etc/kubernetes/node-feature-discovery/nfd-master.conf
+        timeout: 30
+        expected_output: |
+          Node Feature Discovery Master
+          configuration file parsed
+          configuration successfully updated
+          starting the nfd api controller
+        error_strings: |
+          panic
+          fatal
+          FATAL
+        post: |
+          # Verify clean shutdown
+          ps aux | grep nfd-master | grep -v grep || true
```

Comment:
Not needed, script already does `term_wait_kill` test under the hood to ensure it properly terminated.
```suggestion
```